data/modules/scripts/gamestore/init.lua

```diff
@@ -1841,13 +1841,16 @@ end
 	@param coins (number) - The amount of coins to be removed.
 	@return (boolean) - Returns true if the coins were successfully removed, false otherwise.
 --]]
-function Player.removeAllCoins(self, coins)
+function Player.removeAllCoins(self, coins, offerCoinType)
 	-- Check if it is possible to remove all the coins.
 	if self:canRemoveAllCoins(coins) then
 		local tibiaCoins = self:getTibiaCoins()
 		-- Check if there are enough Tibia coins to remove.
-		if tibiaCoins >= coins then
+		if tibiaCoins >= coins and (offerCoinType == GameStore.CoinType.Coin or offerCoinType == GameStore.CoinType.Transferable) then
```

Comment:
Essa parte está correta de fato? No caso, a segunda condição ( o or) deveria ser checado aqui? Já que não deve remover o coin "normal" se for transferable?

Reply:
Aparentemente, a segunda condição (o `or`) está correta nesse contexto. Se a variável  `offerCoinType` for igual a  `GameStore.CoinType.Coin` ou igual a `GameStore.CoinType.Transferable`, a condição será considerada verdadeira. Isso significa que, se a condição for atendida, o código prosseguirá para a próxima instrução.

No caso específico mencionado, a condição  `offerCoinType == GameStore.CoinType.Coin` permite que a transação seja concluída se a moeda oferecida for do tipo "Coin". Por outro lado, a condição  `offerCoinType == GameStore.CoinType.Transferable` também permite a transação se a moeda oferecida for do tipo "Transferable". Assim, a condição  `or` garante que a transação seja autorizada se a moeda for de qualquer um desses dois tipos.

Em resumo, se  `tibiaCoins ` for maior ou igual a  `coins` e a moeda oferecida for do tipo "Coin" ou "Transferable", a condição será verdadeira e a ação correspondente poderá ser executada.

Reply:
Sim, essa parte eu entendi. O que eu quero dizer é que no "or" está checando se a moeda é "transverível", mas o if em questão é apenas para remover moeda não transferível, então nunca vai entrar no próximo if (se entendi bem).

Reply:
Se a intenção é remover apenas moedas não transferíveis e ignorar as moedas transferíveis, então a condição `offerCoinType == GameStore.CoinType.Transferable` no `or` é incorreta. Essa condição permitiria a entrada no próximo `if` mesmo se a moeda fosse transferível.